src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/SQLClause.java

```diff
@@ -81,6 +87,10 @@ else if (parent instanceof MySqlSelectQueryBlock) {
         else {
             throw new IllegalStateException("Unsupported place to use nested field under parent: " + parent);
         }
+
+        if (path != null) {
+            nestedFunc.getParameters().add(1, new SQLCharExpr(path));
```

Comment:
why the SQLCharExpr is added to the specified position in the list? is there any test cases to cover this?

Reply:
also, please do not just put `1` constant, assign it to a local final variable with descriptive name, something like `final int pathParameterIndex = 1;`

Reply:
@arsen-es Done.

@penghuo The unittests [here](https://github.com/opendistro-for-elasticsearch/sql/blob/master/src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/NestedFieldRewriterTest.java) goes through AST match check.